### PR TITLE
Add math min/max for Float and Value

### PR DESCRIPTION
**Pull request by winksaville**
Needed to easily do:
```
        for (pSelf.zbuffer) |*elem| {
            elem.* = math.maxValue(@typeOf(elem.*));
        }
```

### CODE REVIEW

File: std/math/index.zig [hunk 1]
```diff
@@ -714,3 +714,92 @@ test "max value type" {
     const x: u32 = maxInt(i32);
     assert(x == 2147483647);
 }
+
+pub fn maxFloat(comptime T: type) T {
+    return switch (T) {
+        f16 => f16_max,
+        f32 => f32_max,
+        f64 => f64_max,
+        else => @compileError("Expecting type to be a float"),
```
Comment (tgschultz) [inline std/math/index.zig]: f128 is a valid float type.

**Comment by winksaville** [comment]
There is no f128 in all of std/math, just a TODO in sqrt.zig. I think
supporting f128 is another PR.

**Comment by daurnimator** [comment]
I don't think the float implementation here makes sense: the min/max float values is negative/positive infinity.

**Comment by andrewrk** [comment]
Hmm, yeah on closer inspection I think I jumped the gun here. I'm going to revert and let's talk about it some more.

**Comment by andrewrk** [comment]
@winksaville I reverted this. Apologies for jumping the gun. Can we talk more about the use case? What did your code example do, and why did it want the min/max float values? why didn't it want true_max or infinity?

**Comment by tgschultz** [comment]
> I don't think the float implementation here makes sense: the min/max float values is negative/positive infinity.

Wouldn't this objection also apply to the `std.math` constants backing this implementation?

**Comment by andrewrk** [comment]
Right, OK, so it's a reserved float value in your application which has special meaning. A couple questions:

 * why would this be maxValue, instead of maxFloat?
 * why would this be maxFloat, instead of `std.math.nan`?

**Comment by winksaville** [comment]
I'm using maxValue as its more generic and works for Floats or Ints. As for initializing to std.math.nan then the compare wouldn't work and, of course wouldn't work for Int's. The value isn't a "reserved float" the comparison necessary as future pixels written might be "in front of" the current value and need to be overwritten.

**Comment by andrewrk** [comment]
Your graphics code is generic across integers and floats?

**Comment by winksaville** [comment]
No, I'm just trying to learn and seems that it could be.

Should I create a new PR?

**Pull request merged**